### PR TITLE
Return 404 instead of 403 when retrieving an event without perms

### PR DESCRIPTION
**Pull request by anoadragon453**
Part of fixing https://github.com/matrix-org/sytest/issues/652

Sytest PR: https://github.com/matrix-org/sytest/pull/667

### CODE REVIEW

File: synapse/rest/client/v1/room.py [hunk 1]
```diff
@@ -568,7 +568,15 @@ def __init__(self, hs):
...
-        event = yield self.event_handler.get_event(requester.user, room_id, event_id)
+        try:
+            event = yield self.event_handler.get_event(
+                requester.user, room_id, event_id
+            )
+        except AuthError as e:
+            # This endpoint is supposed to return a 404 when the requester does
+            # not have permission to access the event
+            # https://matrix.org/docs/spec/client_server/r0.5.0#get-matrix-client-r0-rooms-roomid-event-eventid
+            return (404, e.msg)
```
Comment (erikjohnston) [inline synapse/rest/client/v1/room.py]: I think this should be a `(404, "Event not found.")`, as the only reason to not return a 403 is to hide the fact the event exists?

Comment (anoadragon453) [reply synapse/rest/client/v1/room.py]: I'm not sure if the original intention in the spec was to hide that fact, but it does seem like a good idea.

**Comment by anoadragon453** [comment]
Blocked on making sure client developers aren't negatively affected by this here: https://matrix.to/#/!lBbDUDayNiOPvtzMev:lain.haus/$15645798291323rbdZX:amorgan.xyz?via=matrix.org&via=disroot.org&via=amorgan.xyz

**Comment by anoadragon453** [comment]
Added an `M_NOT_FOUND` errcode to it as of https://github.com/matrix-org/matrix-doc/pull/2204.

**Comment by anoadragon453** [comment]
@erikjohnston Could I have another review since there was a functionality change?

**Comment by erikjohnston** [comment]
Might be worth changing the other place we return a 404 if there is a reason to use a `SynapseError`?

**Comment by anoadragon453** [comment]
Have updated. The reasoning behind using a `SynapseError` was to send an `M_NOT_FOUND` alongside it.

**Pull request merged**